src/browser/board.js

```diff
@@ -0,0 +1,170 @@
+import CONSTANTS from './constants.js';
+
+function createMatrix(size) {
+  createMatrixBrowser(size);
+  return createMatrixConsole(size);
+}
+
+function createMatrixBrowser(size) {
+  let body = createBody(size);
+  document.getElementsByTagName('body')[0].innerHTML = body;
+}
+
+function createBody(size) {
+  let body = `<div id='board' class='board'> \n`;
+  for (let lineIndex = 0; lineIndex < size; lineIndex += 1) {
+    body += `<div id='line ` + lineIndex + `' class='line'>\n`;
+    for (let columnIndex = 0; columnIndex < size; columnIndex += 1) {
+      body += createCell(lineIndex, columnIndex, size);
+    }
+    body += `</div>\n`;
+  }
+  body += `<div id='winningBar'></div>\n</div>\n<div id='play-again-button'></div>\n`;
+  return body;
+}
+
```

Comment:
Vou aproveitar aqui pra falar de um ponto que a gente tem frisado bastante, que é que usamos muito pouco (ou talvez nenhum) for loops assim no nosso código.

Como ficaria a função createBody seguindo mais as nossos padrões. (exemplos ilustrativos, não sei se vai rodar se copiar direto. E precisei mudar a função pra receber a matrix)
```js
// Tradução mais direta
function createBody(matrix) {
  let body = `<div id='board' class='board'> \n`;
  matrix.forEach((line, lineIndex) => {
    body += `<div id='line ` + lineIndex + `' class='line'>\n`;
    line.forEach((column, columnIndex) => {
      body += createCell(lineIndex, columnIndex, size);
    });
    body += `</div>\n`;
  })
  body += `<div id='winningBar'></div>\n</div>\n<div id='play-again-button'></div>\n`;
  return body;
}

// Uma versão um pouco mais avançada

function createBody(matrix) {
  const linesHtml = matrix.map((line, lineIndex) => createLine(line, lineIndex))
  return `<div id='board' class='board'> 
    ${linesHtml} 
    <div id='winningBar'></div>
  </div>
  <div id='play-again-button'></div>`;
}

function createLine(line, lineIndex) {
  const size = line.length;
  const cellsHtml = line.map((column, columnIndex) => createCell(lineIndex, columnIndex, size)).join('');
  return `<div id='line ` + lineIndex + `' class='line'>
    ${cellsHtml}
  </div>`;
}
```

Reply:
O principal argumento a favor desses métodos é que a tradução mental do código fica mais fácil, depois que você se habitua à sintaxe. A gente passa a pensar mais em  "a cada linha / coluna faça isso" ao invés "pra um indice i, sendo i variando entre 0 e um tamanho, pega o elemento com indice [i].... "